src/components/navigation-header/index.tsx

```diff
@@ -1,158 +1,223 @@
+ * the headers under `/{productSlug}` pages.
+ */
+const PRODUCT_SLUGS_TO_LOGOS = {
+  vault: VaultLogo,
+  waypoint: WaypointLogo,
+}
+
+const HomePageHeaderContent = () => {
   const betaProductSlugs = __config.dev_dot.beta_product_slugs
+
+  return (
+    <div className={s.leftSide}>
+      <div className={s.contentBeforeNav}>
+        <InlineSvg
+          className={s.siteLogo}
+          src={require('./img/logo-white.svg?include')}
+        />
+      </div>
+      <nav className={s.nav}>
+        <ul className={s.navList}>
+          {HOME_PAGE_NAV_ITEMS.map(
+            (navItem: { label: string; pathSuffix: string }) => {
+              const { label, pathSuffix } = navItem
+              const menuItems = betaProductSlugs.map((slug: ProductSlug) => ({
+                icon: slug,
+                label: productSlugsToNames[slug],
+                path: `/${slug}/${pathSuffix}`,
+              }))
+              return (
+                <li key={pathSuffix}>
+                  <NavigationHeaderDropdownMenu
+                    itemGroups={[menuItems]}
+                    label={label}
+                  />
+                </li>
+              )
+            }
+          )}
+        </ul>
+      </nav>
+    </div>
+  )
+}
+
+const ProductPageHeaderContent = () => {
   const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
   const currentProduct = useCurrentProduct()
-
-  const Header = ({ children }: { children: ReactNode }) => (
-    <header className={s.root}>{children}</header>
+  const isBetaProduct = useIsBetaProduct(currentProduct.slug)
+  const betaProductSlugs = __config.dev_dot.beta_product_slugs
+  const companyLogo = (
+    <InlineSvg className={s.companyLogo} src={HashiCorpLogo} />
   )
+  const homeMenuItem = { icon: 'home', label: 'Developer Home', path: '/' }
+  const betaProductMenuItems = betaProductSlugs.map((slug: ProductSlug) => ({
+    icon: slug,
+    label: productSlugsToNames[slug],
+    path: `/${slug}`,
+  }))
+  const allMainMenuItems = [[homeMenuItem], betaProductMenuItems]
+  const productLogo = PRODUCT_SLUGS_TO_LOGOS[currentProduct.slug]
+  const isProductHomePage = currentPath === `/${currentProduct.slug}`
+
+  return (
+    <div className={s.leftSide}>
+      <div className={s.contentBeforeNav}>
+        <NavigationHeaderDropdownMenu
+          ariaLabel="Main menu"
+          buttonClassName={s.companyLogoMenuButton}
+          itemGroups={allMainMenuItems}
+          leadingIcon={companyLogo}
+        />
+        <Link href={`/${currentProduct.slug}`}>
+          <a
+            aria-current={isProductHomePage ? 'page' : undefined}
+            aria-label={`${currentProduct.name} home`}
+            className={s.productLogoLink}
+          >
+            <InlineSvg className={s.productLogo} src={productLogo} />
+          </a>
+        </Link>
+      </div>
+      {isBetaProduct && (
+        <nav className={s.nav}>
```

Comment:
Have you considered extracting this into its own component?

Reply:
Are you thinking of one that would render something like this that both `HomePageHeaderContent` and `ProductPageHeaderContent` could use and pass the `<li>` elements to as `children`?

```tsx
const SharedNavElement = ({ children }) => {
  return (
    <nav className={s.nav}>
      <ul className={s.navList}>
        {children}
      </ul>
    </nav>
  )
}
```